### PR TITLE
[Merged by Bors] - feat: in a C⋆-algebra, `b⁻¹ ≤ a⁻¹ ↔ a ≤ b` for nonnegative `a b : Aˣ`

### PR DESCRIPTION
**Pull request by j-loreaux**
---

- [x] depends on: #16050
- [x] depends on: #16266
- [x] depends on: #16336 
- [x] depends on: #16337 
- [x] depends on: #16339 
- [x] depends on: #16340
- [x] depends on: #16343

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by dupuisf** [comment]
I think it would nice to get a version about `fun x => x ^ (-1 : ℝ)` instead `inv`, with only an invertibility condition for the smallest one.

**Comment by j-loreaux** [comment]
Let me know if this is enough. I was going to add more things, but I think I just want to wait until we have strict positivity. The other unbundled statements just didn't seem so nice to me right now.

File: Mathlib/Analysis/CStarAlgebra/ContinuousFunctionalCalculus/Order.lean [hunk 1]
```diff
@@ -189,6 +191,122 @@ lemma nnnorm_le_natCast_iff_of_nonneg (a : A) (n : ℕ) (ha : 0 ≤ a := by cfc_
...
+lemma CStarRing.inv_le_one_iff_one_le_inv {a : Aˣ} (ha : 0 ≤ (a : A)) :
+    (↑a⁻¹ : A) ≤ 1 ↔ 1 ≤ a := by
+  simpa using CStarRing.inv_le_iff ha (b := 1) (by simp)
+
+lemma CStarRing.inv_le_one {a : Aˣ} (ha : 1 ≤ a) : (↑a⁻¹ : A) ≤ 1 :=
+  CStarRing.inv_le_one_iff_one_le_inv (zero_le_one.trans ha) |>.mpr ha
+
+lemma CStarRing.le_one_of_inv_le_one {a : Aˣ} (ha : 1 ≤ (↑a⁻¹ : A)) : (a : A) ≤ 1 := by
+  simpa using CStarRing.inv_le_one ha
+
```
Comment (dupuisf) [inline Mathlib/Analysis/CStarAlgebra/ContinuousFunctionalCalculus/Order.lean:295]:
```suggestion
lemma CStarRing.le_one_of_one_le_inv {a : Aˣ} (ha : 1 ≤ (↑a⁻¹ : A)) : (a : A) ≤ 1 := by
  simpa using CStarRing.inv_le_one ha

```

**Review by dupuisf — COMMENTED** [review]
Looks good!

bors d+

**Comment by j-loreaux** [comment]
bors merge

**Pull request closed without merging**